src/network.rs

```diff
@@ -433,7 +446,19 @@ impl EventLoop {
 						num_established,
 						cause,
 					} => {
-						trace!("Connection closed. PeerID: {:?}. Endpoint: {:?}. Num establ: {:?}. Cause: {:?}", peer_id, endpoint, num_established, cause);
+						trace!("Connection closed. PeerID: {:?}. Address: {:?}. Num establ: {:?}. Cause: {:?}", peer_id, endpoint.get_remote_address(), num_established, cause);
+
+						if let Some(cause) = cause {
+							match cause {
+								// remove peer with failed connection
+								ConnectionError::IO(_) | ConnectionError::Handler(_) => {
+									self.swarm.behaviour_mut().kademlia.remove_peer(&peer_id);
```

Comment:
We should prob check for other open connections first, as there could be a case of partial failure when some connections are dropped and some are not...

Reply:
agree, but for the time being, until we implement a retry mechanism, this will have to suffice